pkg/cli/alpha/config-gen/cmd.go

```diff
@@ -17,75 +17,99 @@ limitations under the License.
+
+	"embed"
 	"fmt"
 	"io/ioutil"
 	"log"
 	"os"
 	"path/filepath"
 	"strings"
 
-	// TODO: switch to embed
-	"github.com/markbates/pkger"
 	"github.com/spf13/cobra"
-
-	// import pkged files
-	_ "sigs.k8s.io/kubebuilder/v3"
 	"sigs.k8s.io/kustomize/kyaml/fn/framework"
+	"sigs.k8s.io/kustomize/kyaml/fn/framework/command"
+	"sigs.k8s.io/kustomize/kyaml/fn/framework/parser"
 	"sigs.k8s.io/kustomize/kyaml/kio"
 	"sigs.k8s.io/kustomize/kyaml/yaml"
 )
 
-// NewCommand returns a new cobra command
-func NewCommand() *cobra.Command {
-	kp := &KubebuilderConfigGen{}
-
-	// legacy kustomize function support
-	legacyPlugin := os.Getenv("KUSTOMIZE_PLUGIN_CONFIG_STRING")
-	err := yaml.Unmarshal([]byte(legacyPlugin), kp)
-	if err != nil {
-		log.Fatal(err)
-	}
-
-	// Eager check to make sure pkged templates are found.
-	err = pkger.Walk("/pkg/cli/alpha/config-gen/templates/resources", func(_ string, _ os.FileInfo, err error) error {
-		return err
-	})
-	if err != nil {
-		// this shouldn't fail if it was compiled correctly
-		log.Fatal(err)
-	}
+// TemplateFS contains the templates used by config-gen
+//go:embed templates/resources/* templates/patches/*
+var TemplateFS embed.FS
 
-	c := framework.TemplateCommand{
-		API: kp,
+func buildProcessor(value *KubebuilderConfigGen) framework.ResourceListProcessor {
+	return framework.TemplateProcessor{
+		MergeResources: true,
 
-		MergeResources: true, // apply additional inputs as patches
-
-		// these are run before the templates
 		PreProcessFilters: []kio.Filter{
 			// run controller-gen libraries to generate configuration from code
-			ControllerGenFilter{KubebuilderConfigGen: kp},
+			ControllerGenFilter{KubebuilderConfigGen: value},
 			// inject generated certificates
-			CertFilter{KubebuilderConfigGen: kp},
+			CertFilter{KubebuilderConfigGen: value},
 		},
 
-		// generate resources
-		// keep casting -- required by pkger to find the directory
-		TemplatesFn: framework.TemplatesFromDir(pkger.Dir("/pkg/cli/alpha/config-gen/templates/resources")),
-
-		// patch resources
-		PatchTemplatesFn: framework.PatchTemplatesFromDir(
-			CRDPatchTemplate(kp),
-			CertManagerPatchTemplate(kp),
-			ControllerManagerPatchTemplate(kp),
-		),
+		ResourceTemplates: []framework.ResourceTemplate{{
+			Templates: parser.TemplateFiles(filepath.Join("templates", "resources")).FromFS(TemplateFS),
+		}},
+		PatchTemplates: []framework.PatchTemplate{
+			&framework.ResourcePatchTemplate{
+				Selector: &framework.Selector{
+					Kinds: []string{"CustomResourceDefinition"},
+					ResourceMatcher: func(m *yaml.RNode) bool {
+						meta, _ := m.GetMeta()
+						return value.Spec.Webhooks.Conversions[meta.Name]
+					},
+				},
+				Templates: parser.TemplateFiles(filepath.Join("templates", "patches", "crd")).FromFS(TemplateFS),
+			},
+			&framework.ResourcePatchTemplate{
+				Selector: &framework.Selector{
+					Kinds:  []string{"Deployment"},
+					Names:  []string{"controller-manager"},
```

Comment:
This used to also have a namespace selector. Is that no longer needed? Since this is no longer in a closure itself, you'd need to use a template string.

Reply:
Done